packages/git-encrypted/src/keyDerivation.ts

```diff
@@ -0,0 +1,13 @@
+import argon2 from 'argon2';
+
+// TODO: read salt from user's repository
+const SALT = Buffer.from('deadbeef');
+
+export const hash = async (password: string) => {
```

Comment:
@johannes87 I imagined that the salt would be provided as an input to the function, as it needs to be unique per repository. I assumed there would be other vectors provided to the `argon2.hash()` function, like `timeCost`, `memoryCost`, and `parallelism`. I’d favour setting those in our codebase rather than using the defaults from the imported code. In that way, if the imported module ever changes, our code doesn’t break.

It might also be good to specify the type of argon2id as that seems to be the most secure option.